src/cppcache/include/geode/Struct.hpp

```diff
@@ -138,7 +138,7 @@ class CPPCACHE_EXPORT Struct : public Serializable {
    * Returns the name of the field corresponding to the index number in the
    * Struct
    */
-  virtual const char* getFieldName(const int32_t index) const;
+  virtual const std::string* getFieldName(const int32_t index) const;
```

Comment:
Yeah, if we consider when field does not exist at index a failure case, then I agree that throwing is better than returning `nullptr` and expecting someone else to handle it or blow up later.